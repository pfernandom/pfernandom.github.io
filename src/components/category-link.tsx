/* eslint-disable react/no-array-index-key */
import { LayoutGroup, motion, useAnimation, Variants } from 'framer-motion';
import Image from 'next/image';
import Link from 'next/link';
import { useRouter } from 'next/router';
import React, { SetStateAction, useEffect, useMemo, useState } from 'react';
import { Callback, debouncer } from 'src/helpers/utils';
import { PostInfo } from 'src/models/data';

function ToParagraphs({
  posts,
  className,
  style,
}: {
  posts: Array<string>;
  className?: string;
  style: Record<string | number, string>;
}) {
  return (
    <div className={className} style={style}>
      {posts?.map((el, index) => (
        <p key={`description-p-${index}`}>{el}</p>
      ))}
    </div>
  );
}

ToParagraphs.defaultProps = { className: '' };

export function useMediaQuery(query) {
  const [matches, setMatches] = useState(false);

  useEffect(() => {
    const media = window.matchMedia(query);
    if (media.matches !== matches) {
      setMatches(media.matches);
    }
    const listener = () => {
      setMatches(media.matches);
    };
    media.addListener(listener);
    return () => media.removeListener(listener);
  }, [matches, query]);

  return matches;
}

function getVariants(isLg: boolean) {
  if (isLg) {
    return {
      cardMotion: {
        rest: {
          scale: 1,
          zIndex: 2,
          transition: {
            type: 'linear',
            duration: 0.2,
            delayChildren: 2,
            when: 'beforChildren',
          },
        },
      },
      titleMotion: {
        rest: {
          opacity: 1,
          y: 1,
          width: '90%',
          borderTopRightRadius: 0,
          scale: 1,
          transition: {
            type: 'linear',
            duration: 0.2,
          },
        },
        hover: {},
      },
      imageMotion: {
        rest: {
          opacity: 0.2,
          transition: {
            type: 'linear',
            duration: 0.5,
          },
        },
        hover: {
          opacity: 0.3,
        },
      },
      descriptionMotion: {
        rest: {
          opacity: 1,
          y: 0,
          width: '100%',
          borderTopRightRadius: 0,
          transition: {
            type: 'linear',
            duration: 0.2,
          },
        },
      },
    };
  }
  const cardMotion = {
    rest: { flexGrow: 1, scale: 1, zIndex: 0, transition: { when: 'afterChildren' } },
    hover: {
      // flexGrow: shouldReduceMotion ? 1 : 2,
      scale: isLg ? 1 : 1.3,
      zIndex: 2,
      transition: {
        type: 'linear',
        duration: 0.2,
        delayChildren: 2,
        when: 'beforChildren',
      },
    },
  };

  const titleMotion: Variants = {
    rest: { y: 50, opacity: 1, borderTopRightRadius: '0.5em', width: '40%' },
    hover: {
      y: 0,
      width: '90%',
      borderTopRightRadius: 0,
      transition: {
        type: 'linear',
        duration: 0.2,
      },
    },
  };

  const imageMotion = {
    rest: { opacity: 1 },
    hover: {
      opacity: 0.3,
      transition: {
        type: 'linear',
        duration: 0.5,
      },
    },
  };

  const descriptionMotion = {
    rest: { opacity: 0, delay: 0 },
    hover: {
      opacity: 1,
      transition: {
        duration: 0.2,
        type: 'tween',
        ease: 'easeIn',
        delay: 0.2,
      },
    },
  };

  return { cardMotion, titleMotion, imageMotion, descriptionMotion };
}

export function CategoryLink({ post }: { post: PostInfo }) {
  const controls = useAnimation();
  const [isHovering, setIsHovering] = useState(false);
  const router = useRouter();

  const isLg = useMediaQuery('(max-width: 700px)');

  const { cardMotion, titleMotion, imageMotion, descriptionMotion } = getVariants(isLg);

  useEffect(() => {
    controls.start(isHovering ? 'hover' : 'rest');
  }, [isHovering, isLg]);

  const debouncedSetIsHovering = debouncer(500, (params: SetStateAction<boolean>) => {
    setIsHovering(params);
  });

  return (
    <motion.div
      key="my_category_link"
      className="category__link"
      variants={cardMotion}
      animate={controls}
      onHoverStart={() => {
        router.prefetch(`/${post.slug}`);
        debouncedSetIsHovering(true);
      }}
      onHoverEnd={() => {
        debouncedSetIsHovering(false);
      }}
    >
      <Link href={`/${post.slug}`} style={{ position: 'relative', display: 'block' }}>
        <motion.div
          style={{ position: 'absolute', zIndex: 3 }}
          animate={controls}
          variants={titleMotion}
          initial="rest"
          className="category__title"
        >
          {post.frontmatter.title}
        </motion.div>
        <div style={{ position: 'relative' }}>
          <motion.div
            variants={imageMotion}
            animate={controls}
            initial="rest"
            className="category__hero-wrapper"
          >
            <Image
              className="category__hero"
              style={{ zIndex: 0 }}
              src={post.frontmatter.hero_image}
              priority
              alt="Picture of the author"
              // When "responsive", similar to "fluid" from Gatsby
              // When "intrinsic", similar to "fluid" with maxWidth from Gatsby
              // When "fixed", similar to "fixed" from Gatsby
              layout="responsive"
              // Optional, similar to "blur-up" from Gatsby
              // placeholder="blur"
              // Optional, similar to "width" in Gatsby GraphQL
              width={500}
              // Optional, similar to "height" in Gatsby GraphQL
              height={isLg ? 300 : 500}
              objectFit="cover"
            />
          </motion.div>

          <motion.div
            className="category__description"
            animate={controls}
            initial="rest"
            variants={descriptionMotion}
          >
            <ToParagraphs style={{ margin: '1em' }} posts={post.frontmatter.description} />
          </motion.div>
        </div>
      </Link>
    </motion.div>
  );
}

export function CategoryLinkSection({ posts }: { posts: Array<PostInfo> }) {
  return (
    <div className="category__list">
      <LayoutGroup>
        {posts.map((post: any) => (
          <CategoryLink key={post?.slug} post={post} />
        ))}
      </LayoutGroup>
    </div>
  );
}

export default CategoryLinkSection;
