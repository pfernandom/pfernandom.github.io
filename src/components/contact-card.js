import React from 'react';
import PropTypes from 'prop-types';
import './contact-card.scss';
import { GithubIcon, TwitterIcon, LinkedInIcon, PhoneIcon, EmailIcon } from './icons';

function ContactCard({ name, role, summary, contact = [], highlights }) {
  return (
    <div className="contact-card">
      <div className="info">
        <h1>{name}</h1>
        <h2>{role}</h2>

        <div className="summary">{summary}</div>
        <ul>
          {highlights.map(h => (
            <li key={h}>{h}</li>
          ))}
        </ul>
      </div>
      <div className="contact-section">
        <h3>Contact</h3>
        <ul>
          {contact.map(c => (
            <li key={c.link} className={c.print ? '' : 'no-print'}>
              {
                {
                  github: <GithubIcon />,
                  twitter: <TwitterIcon />,
                  linkedin: <LinkedInIcon />,
                  phone: <PhoneIcon />,
                  mail: <EmailIcon />,
                }[c.icon]
              }
              <a href={c.link} target="_blank" rel="noopener noreferrer">
                {c.text}
              </a>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}

ContactCard.propTypes = {
  name: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired,
  contact: PropTypes.array,
  highlights: PropTypes.arrayOf(PropTypes.string),
};

ContactCard.defaultProps = {
  contact: [],
  highlights: [],
};

export default ContactCard;
