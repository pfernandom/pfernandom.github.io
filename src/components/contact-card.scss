.contact-card{
    min-height: 10em;
    display: flex;

    .info {
        display: flex;
        flex-wrap: wrap;
        align-items: flex-start;
        align-content: baseline;
        h1 {
            flex-basis: 50%;
        }

        h2{
            font-size: 1.5em;
            color: grey;
            margin-top: 0px;
        }

        p {
            margin-bottom: 0.5em;
        }
    }
    
    .contact-section {
        font-size: small;
    }

    .summary {
        flex-basis: 100%;
    }

    .icon{
        width: 20px;
        font-size: 1.5em;
        display:inline-block;
        margin:4px;
        margin-right: 10px;
        color:#C2C2C2;
    }

    ul{
        list-style-type: none;
        margin: 0.5em 0;
        li {
            display: flex;
            margin-bottom: 0;
        }
        
        a{
            height: 20px;
            margin: 5px;
            display:inline-block;
            vertical-align: bottom;
            color: #0067e4;
            text-decoration: dotted;
        }
    }

    @media (max-width: 770px) {
        &.contact-card{
            margin:0px;
            padding: 0px;
            margin: 0.5em;
            flex-direction: column;
        }
        .contact-section{
            margin:0px;
            padding: 0px;
        }

        .icon{
            margin:5px;
            color:#C2C2C2;
        }
        ul{
            padding: 0;
            display: flex;
            flex-wrap: wrap;
        }
        li{
            display:inline-block;
            margin: 0.5em;
        }
    }

    @media print{
        h1 {
            font-size: 2em;
        }
        .icon{
            margin:5px;
            color:#C2C2C2;
        }

        &.contact-card{
            display: flex;
            flex-direction: row;
            ul {
                li {
                    margin: 0.2em;
                }
            }
        }

        .contact-section {
            font-size: small;
            li {
                height:3em;
            }
        }
    }
}