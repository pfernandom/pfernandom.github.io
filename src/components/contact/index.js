import React from 'react'
import './ContactCard.scss'

class ContactCard extends React.Component {
	render() {
		return (
			<div className="contact-card">
				<h1>Pedro Marquez</h1>
				<h2>Full-stack Software Developer</h2>
				
				<p>
					Full Stack developer, Java Architect and Certified Ethical Hacker. 
				</p>
				<ul>
					<li>5+ years of experience in Java technologies (back-end).</li>
					<li>7+ years of experience in front-end technologies (JavaScript, CSS, HTML).</li>
					<li>2 years of experience in Application Security and Penetration Testing.</li>
					<li>Scrum Master trained.</li>
					<li>Six Sigma Green Belt trained.</li>
				</ul>

			</div>
		)
	}
}

export default ContactCard