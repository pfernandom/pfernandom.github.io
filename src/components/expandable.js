import React from 'react';
import PropTypes from 'prop-types';
import './expandable.scss';

export default class Expandable extends React.Component {
  constructor(...props) {
    super(...props);
    this.state = {
      expanded: null,
    };
  }

  componentDidUpdate({ expanded: prevExpanded }) {
    const { expanded } = this.props;
    if (expanded !== prevExpanded) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        expanded,
      });
    }
  }

  onToggle(ev) {
    // This is needed because pouter <details> elements were
    // capturing the inner details event
    ev.stopPropagation();
    const { onToggle } = this.props;
    if (onToggle) {
      onToggle.call(this, ev);
    }
    this.setState({
      expanded: ev.target.open,
    });
  }

  toggle() {
    const { expanded } = this.state;
    this.setState({
      expanded: !expanded,
    });
  }

  render() {
    const { expanded: expandedInState } = this.state;
    const {
      title,
      theme,
      expanded: expandedInProps,
      children,
      preview,
      buttonClassName,
    } = this.props;
    const expanded = expandedInState == null ? expandedInProps : expandedInState;
    return (
      <React.Fragment>
        <details open={expanded} onToggle={this.onToggle.bind(this)}>
          <summary
            className={`expandable__title 
            ${theme === 'inline' && 'expandable__title--inline'}
            ${buttonClassName}
            `}
          >
            <h3
              className={`expandable__heading 
            ${theme === 'inline' && 'expandable__heading--inline'}`}
            >
              {title}
            </h3>
          </summary>

          <div
            className={`expandable__content
            ${theme === 'inline' && 'expandable__content--inline'}
            ${expanded ? '' : 'no-print'}`}
            hidden={!expanded}
          >
            {children}
          </div>
        </details>
        {preview && (
          <div className={`expandable__preview ${expanded ? 'no-print' : ''}`} hidden={expanded}>
            {preview.call(this, { toggle: this.toggle.bind(this) })}
          </div>
        )}
      </React.Fragment>
    );
  }
}
Expandable.propTypes = {
  title: PropTypes.PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  theme: PropTypes.string,
  buttonClassName: PropTypes.string,
  expanded: PropTypes.bool,
  onToggle: PropTypes.func,
  preview: PropTypes.func,
  children: PropTypes.node.isRequired,
};

Expandable.defaultProps = {
  expanded: false,
  buttonClassName: '',
  theme: '',
  onToggle: () => {},
  preview: () => {},
};
