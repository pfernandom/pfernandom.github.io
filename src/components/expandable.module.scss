.heading {
    display: flex;
    width: 100%;
    padding: 0;
    margin: 0;
}

.headingButton {
    border: none;
    background-color: #6895cc;
    color: #fdfdfd;
    padding: .5rem;
    text-align: left;
    font-size: 1.4rem;
    display: flex;
    width: 100%;
    justify-content: space-between;
    align-items: center;
    &:focus {
        background-color: #4c6d96;
    }
}

.inlineHeadingButton {
    display: flex;
    padding: 5px;
    align-items: center;
}

.icon {
    width: 0.5em;
    margin: 5px;
}

.content {
    display:flex;
    flex-wrap: wrap;
    &[hidden] {
        display:none;
    }
}


@media print{
    [hidden] {
        .responsibilities {
            display: hidden;
        }
    }
    .content {
        display: initial;
        &[hidden] {
            display:initial;
        }
    }

    .headingButton {
        border: 1px solid #d4d4d4;
    }

    .inlineHeadingButton{
        border: none;
    }
}