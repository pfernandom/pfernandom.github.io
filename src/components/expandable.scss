.expandable__heading {
    display: flex;
    width: 100%;
    padding: 0;
    margin: 0;
}

.expandable__title {
    border: none;
    background-color: #6895cc;
    color: #fdfdfd;
    padding: .5rem;
    text-align: left;
    display: flex;
    justify-content: space-between;
    align-items: center;
    &:focus {
        background-color: #4c6d96;
    }
}

.expandable__content {
    display:flex;
    flex-wrap: wrap;
    border-radius: 0 0 5px 5px;
    &[hidden] {
        display:none;
    }
}

.expandable__title--inline {
    align-items: center;
    background-color: #f1f1f1;
    border-radius: 5px 5px 0 0;
    color: rgba(0, 0, 0, 0.8);
}

.expandable__heading--inline {
    font-size: 1em;
    font-weight: 200;
}

.expandable__content--inline {
    background-color: #fff6e2;
}

.expandable__preview {
    margin: 0;
}

@media print{
    [hidden] {
        .responsibilities {
            display: hidden;
        }
    }
    .expandable__content {
        display: initial;
        &[hidden] {
            display:initial;
        }
    }

    .expandable__title {
        border: 1px solid #d4d4d4;
    }

    .expandable__title--inline{
        border: none;
    }
}