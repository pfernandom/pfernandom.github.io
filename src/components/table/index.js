import SCTable from './SCTable'

export {SCTable}