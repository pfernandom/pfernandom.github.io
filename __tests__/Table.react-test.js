import React from 'react';
import renderer from 'react-test-renderer';
import {SCTable} from '../src/components/table';

test('SCTable matches snapshot', () => {
	const component = renderer.create(
		<SCTable>
			<tbody>
				<tr><td>Test</td></tr>
			</tbody>
		</SCTable>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});



