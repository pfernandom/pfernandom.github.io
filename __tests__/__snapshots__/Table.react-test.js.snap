exports[`test SCTable matches snapshot 1`] = `
<table
  className="scTable">
  <tbody>
    <tr>
      <td>
        Test
      </td>
    </tr>
  </tbody>
</table>
`;
